### PR TITLE
temp fix to unmarshalling error in recordSummaryAnnotations

### PR DESCRIPTION
**Pull request by ramessesii2**
Latest deployment of Pipelines-as-Code with OSP 1.13 adds the following annotation the the PipelineRuns spawned by it.
```yaml
results.tekton.dev/recordSummaryAnnotations: >-   {"repo":"actions","commit":"64537b83aba9a0b3110f146dfeb0c4560b69b81d","eventType":"pull_request","pull_request-id":3}
```
 

which leads to this results error reconciling PAC generated pipelineruns:

 ```json
{"level":"error","ts":"2024-03-04T10:33:15.841Z","logger":"watcher","caller":"controller/controller.go:566","msg":"Reconcile error","knative.dev/traceid":"10488519-927a-45f3-9b02-1cad3432e97c","knative.dev/key":"jkhelil-tenant/actions-on-pull-request-gw65r","duration":"378.894458ms","error":"error upserting record: error parsing annotation results.tekton.dev/recordSummaryAnnotations: json: cannot unmarshal number into Go value of type string","stacktrace":"knative.dev/pkg/controller.(*Impl).handleErr\n\t/opt/app-root/src/vendor/knative.dev/pkg/controller/controller.go:566\nknative.dev/pkg/controller.(*Impl).processNextWorkItem\n\t/opt/app-root/src/vendor/knative.dev/pkg/controller/controller.go:543\nknative.dev/pkg/controller.(*Impl).RunContext.func3\n\t/opt/app-root/src/vendor/knative.dev/pkg/controller/controller.go:491"}
```
 
and that prevents log storage and deletion of said pipelineruns.

    - related to PLNSRVCE-1679

Signed-off-by: Satyam Bhardwaj <sabhardw@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

### CODE REVIEW

File: pkg/watcher/results/results.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func (c *Client) ensureResult(ctx context.Context, o Object, opts ...grpc.CallOp
 
 	if topLevel {
 		if value, found := o.GetAnnotations()[annotation.RecordSummaryAnnotations]; found {
-			recordSummaryAnnotations, err := parseAnnotations(annotation.RecordSummaryAnnotations, value)
+			recordSummaryAnnotations, err := parseRecordSummaryAnnotations(annotation.RecordSummaryAnnotations, value)
 			if err != nil {
```
Comment (gabemontero) [inline pkg/watcher/results/results.go:145]: don't you want to check for PermanentError here and ignore it

**Review by gabemontero — CHANGES_REQUESTED** [review]
I don't see how this fixes the error @ramessesii2 

you are still returning an error that is bubbled up to the caller regardless 
and while knative controller will ignore it i.e. what I see in vendor/knative.dev/pkg/controller/controller.go, you still bypass all the subsequent processing that we want to do

@sayan-biswas @avinal would you all agree, or what am I missing

File: pkg/watcher/results/results.go (hunk 1 above)
Comment (ramessesii2) [reply pkg/watcher/results/results.go:145]: I'm not sure why originally the error is not subdued. A little bit of low level detail though - this recordSummaryAnnotaion is a key in the RESULTS table in the database, so every entry/row has this `recordsummary_annotations` filed with the value same as the value of "results.tekton.dev/recordSummaryAnnotations".

So, I'm happy to make that change if it's not critical to the operation of the function. I'm gonna tag @sayan-biswas if he can share any insights.

**Comment by ramessesii2** [comment]
> I don't see how this fixes the error 

@gabemontero I've tried addressing this particular error in logs `error parsing annotation results.tekton.dev/recordSummaryAnnotations: json: cannot unmarshal number into Go value of type string`, this is it's [source from PaC](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/kubeinteraction/resultsannotation.go#L15), the 
`pull_request-id` is an int, results expects a map[string]string. So, in this temp fix, I've basically done type conversion, first unmarshalling into `map[string]any` (just looked online, it's suggested to use interface{} instead) and then to the format Results expect. 
Does that make sense?

**Comment by gabemontero** [comment]
> > I don't see how this fixes the error
> 
> @gabemontero I've tried addressing this particular error in logs `error parsing annotation results.tekton.dev/recordSummaryAnnotations: json: cannot unmarshal number into Go value of type string`, this is it's [source from PaC](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/kubeinteraction/resultsannotation.go#L15), the `pull_request-id` is an int, results expects a map[string]string. So, in this temp fix, I've basically done type conversion, first unmarshalling into `map[string]any` (just looked online, it's suggested to use interface{} instead) and then to the format Results expect. Does that make sense?

OK so you are trying to fix this, not just ignore the error.  I did not pick up on that before.

From what I hear in the slack thread, this is needs to be a permanent solution, and not a temp fix.  That there is not a change needed on the PAC side.

I know we cannot make PAC a dependency, but I'm almost inclined to just duplicate the `ResultAnnotation` struct declaration in Results and just map to that.

Let's get a PR upstream going as well, pull in @sayan-biswas @khrm etc. and sort this out.  The see about temporarily duplicating in this PR short term if it is taking too long to get the upstream fix.

**Comment by ramessesii2** [comment]
Closing it in favor of https://github.com/tektoncd/results/pull/722

**Pull request closed without merging**